src/storage/compression/compression_float.cpp

```diff
@@ -0,0 +1,212 @@
+                          auto value = reinterpret_cast<const T*>(srcBuffer)[i];
+                          return (nullMask && nullMask->isNull(i)) ||
+                                 canUpdateInPlace(std::span(&value, 1), metadata, localUpdateState);
+                      })));
+
+    std::vector<EncodedType> integerEncodedValues(numValues);
+    for (size_t i = 0; i < numValues; ++i) {
+        const size_t posInSrc = i + srcOffset;
+
+        const auto floatValue = reinterpret_cast<const T*>(srcBuffer)[posInSrc];
+        const EncodedType encodedValue = alp::AlpEncode<T>::encode_value(floatValue,
+            metadata.floatMetadata().fac, metadata.floatMetadata().exp);
+        integerEncodedValues[i] = encodedValue;
+    }
+
+    encodedFloatBitpacker.setValuesFromUncompressed(
+        reinterpret_cast<const uint8_t*>(integerEncodedValues.data()), 0, dstBuffer, dstOffset,
+        numValues, metadata.getChild(BITPACKING_CHILD_IDX), nullMask);
+}
+
+template<std::floating_point T>
+BitpackInfo<typename FloatCompression<T>::EncodedType> FloatCompression<T>::getBitpackInfo(
+    const CompressionMetadata& metadata) {
+    return decltype(encodedFloatBitpacker)::getPackingInfo(metadata.getChild(BITPACKING_CHILD_IDX));
+}
+
+template<std::floating_point T>
+bool FloatCompression<T>::canUpdateInPlace(std::span<const T> value,
+    const CompressionMetadata& metadata,
+    CompressionMetadata::InPlaceUpdateLocalState& localUpdateState,
+    const std::optional<common::NullMask>& nullMask, uint64_t nullMaskOffset) {
+    size_t newExceptionCount = 0;
+    std::vector<EncodedType> encodedValues(value.size());
+    const auto bitpackingInfo =
+        decltype(encodedFloatBitpacker)::getPackingInfo(metadata.getChild(BITPACKING_CHILD_IDX));
+    for (size_t i = 0; i < value.size(); ++i) {
+        if (nullMask && nullMask->isNull(nullMaskOffset + i)) {
+            continue;
+        }
+
+        const auto floatValue = value[i];
+        const EncodedType encodedValue = alp::AlpEncode<T>::encode_value(floatValue,
+            metadata.floatMetadata().fac, metadata.floatMetadata().exp);
+        const double decodedValue = alp::AlpDecode<T>::decode_value(encodedValue,
+            metadata.floatMetadata().fac, metadata.floatMetadata().exp);
+        if (floatValue != decodedValue) {
+            ++newExceptionCount;
+            encodedValues[i] = bitpackingInfo.offset;
+        } else {
+            encodedValues[i] = encodedValue;
+        }
+    }
+    localUpdateState.floatState.newExceptionCount += newExceptionCount;
+    // TODO(Royi) hack: if we rewrite the entire chunk we don't take into account previous
+    // exceptions
+    const size_t totalExceptionCount =
+        (value.size() == common::StorageConstants::NODE_GROUP_SIZE) ?
+            localUpdateState.floatState.newExceptionCount :
+            metadata.floatMetadata().exceptionCount + localUpdateState.floatState.newExceptionCount;
```

Comment:
Currently have this hack since during checkpoint (in `Column::checkpointColumnChunkInPlace`) we pass in entire column chunks.  Thus `totalExceptionCount` would always be double the actual exception count, making it always greater than `exceptionCapacity` and so `canUpdateInPlace` would always return false. 

This hack also won't work if we have multiple checkpoint states (e.g. due to insertion + update) as numValues will be split between the checkpoint states.

This won't be a problem anymore once we only pass in updated values during checkpoint so just leaving a TODO here

Reply:
updated in https://github.com/kuzudb/kuzu/pull/3994/files#r1705671328